common/fe-ste-models.go

```diff
@@ -1073,7 +1073,7 @@ func (i *InvalidMetadataHandleOption) UnmarshalJSON(b []byte) error {
 
 // //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 const (
-	DefaultBlockBlobBlockSize      = 8 * 1024 * 1024
+	DefaultBlockBlobBlockSize      = 256 * 1024 * 1024
```

Comment:
I did some perf testing and increasing the default size makes the copy command run almost 3 seconds longer. I plan on reverting this change.

Reply:
@siminsavani-msft What tests did you run? Did you run against multiple different sizes of file and run multiple times?

Reply:
I used the tool-test-perf script to test this which runs azcopy 5 x old and new (total 10). I'm going to add the excel sheet in the following comment to keep the conversation cleaner.

Reply:
The larger chunk size is likely to shine in larger file sizes, can you increase the size of file to the GBs as well and see how it looks

Reply:
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/SIMINS~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/SIMINS~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>
<!--table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
@page
	{margin:.75in .7in .75in .7in;
	mso-header-margin:.3in;
	mso-footer-margin:.3in;}
tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
br
	{mso-data-placement:same-cell;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:black;
	font-size:11.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:"Aptos Narrow", sans-serif;
	mso-font-charset:0;
	mso-number-format:General;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;
	mso-protection:locked visible;
	white-space:nowrap;
	mso-rotate:0;}
-->
</style>
</head>

<body link="#467886" vlink="#96607D">


1 | new | 7 |   | 1 GB file |  
-- | -- | -- | -- | -- | --
1 | current | 25 |   | average new time | 7.2
2 | new | 9 |   | average current time | 8
2 | current | 7 |   |   |  
3 | new | 6 |   |   |  
3 | current | 3 |   |   |  
4 | new | 7 |   |   |  
4 | current | 3 |   |   |  
5 | new | 7 |   |   |  
5 | current | 2 |   |   |  
  |   |   |   |   |  
1 | new | 9 |   | 5 GB file |  
1 | current | 45 |   | average new time | 8.6
2 | new | 11 |   | average current time | 12.8
2 | current | 5 |   |   |  
3 | new | 7 |   |   |  
3 | current | 4 |   |   |  
4 | new | 7 |   |   |  
4 | current | 5 |   |   |  
5 | new | 9 |   |   |  
5 | current | 5 |   |   |  
  |   |   |   |   |  
1 | new | 7 |   | 11 GB file |  
1 | current | 47 |   | average new time | 7.6
2 | new | 7 |   | average current time | 24.8
2 | current | 49 |   |   |  
3 | new | 10 |   |   |  
3 | current | 19 |   |   |  
4 | new | 7 |   |   |  
4 | current | 5 |   |   |  
5 | new | 7 |   |   |  
5 | current | 4 |   |   |  



</body>

</html>

Reply:
As you said, larger chunk size is likely to shine in larger files. I think we should discuss this with the broader team because we may have customers that are transferring a bunch of small files and the 256 MB default may slow them down as those extra seconds can add up. We could do two things: 1. we can make AzCopy recognize that the source size is as large as 1 GB (or other threshold) and use 256 MB as a default and for file sizes smaller than that, AzCopy uses 8 MB default; 2. Leave the default as 8 MB and allow customers to modify this size via the existing --block-blob-size flag. 

I tried transferring 3 256 MB files and this is what I got:

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/SIMINS~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/SIMINS~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>
<!--table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
@page
	{margin:.75in .7in .75in .7in;
	mso-header-margin:.3in;
	mso-footer-margin:.3in;}
tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
br
	{mso-data-placement:same-cell;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:black;
	font-size:11.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:"Aptos Narrow", sans-serif;
	mso-font-charset:0;
	mso-number-format:General;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;
	mso-protection:locked visible;
	white-space:nowrap;
	mso-rotate:0;}
-->
</style>
</head>

<body link="#467886" vlink="#96607D">


1 | new | 7 |   | 256 MB x 3 files |  
-- | -- | -- | -- | -- | --
1 | current | 3 |   | average new time | 6.2
2 | new | 5 |   | average current time | 3.2
2 | current | 3 |   |   |  
3 | new | 5 |   |   |  
3 | current | 4 |   |   |  
4 | new | 7 |   |   |  
4 | current | 3 |   |   |  
5 | new | 7 |   |   |  
5 | current | 3 |   |   |  



</body>

</html>

Reply:
Discussed offline: leave default as is.